File: playgroundJ/src/main/java/com/gluszczykk/playgroundj/_0wstep/Wstep.java
package com.gluszczykk.playgroundj._0wstep;

public class Wstep {
    public static void main(String[] args) {
        String imie = "Krzysiek";
        System.out.println("Hello world! " + ":" + imie + " " + dodaj(2,3));

        Student student1 = new Student("Radek",35 , "Wrocław");
        Student student2 = new Student("Radek",35 , "Wrocław");
        boolean czyTenSam = student1 == student2;
        System.out.println("Czy ten sam student:" +czyTenSam + student1);
    }

    public static int dodaj(int a, int b){
        return a+b;
    }
}

File: playgroundK/src/main/java/com/gluszczykk/playgroundk/_0wstep/wstep.kt
package com.gluszczykk.playgroundk._0wstep

fun main() {
    val imie = "Krzysiek"
    var imie2 = imie
    imie2 = "Dorota"
    println("Hello world from the Kotlin side! :$imie ${dodaj(b=2, a=3)}")

    val student1 = Student(imie= "Radek", miejsceZamieszkania = "Wrocław", wiek =35)
    val student2 = Student("Radek", "Wrocław", wiek =36)
    val student3 = Student("Radek")

    println("Czy ten sam: ${student1 == student2} $student1")
}

fun dodaj(a : Int, b: Int) =  a + b

data class Student(val imie: String, val miejsceZamieszkania: String="Warszawa", val wiek: Int= 18){
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Student

        if (imie != other.imie) return false
        if (miejsceZamieszkania != other.miejsceZamieszkania) return false

        return true
    }

    override fun hashCode(): Int {
        var result = imie.hashCode()
        result = 31 * result + miejsceZamieszkania.hashCode()
        return result
    }
}